File: src/main/java/org/kite9/diagram/model/position/RectangleRenderingInformation.java
package org.kite9.diagram.model.position;

import org.apache.commons.math.fraction.BigFraction;
import org.kite9.diagram.common.objects.OPair;

/**
 * Contains details of how to render a rectangle on screen, possibly containing some 
 * text.
 * 
 * 
 * @author robmoffat
 *
 */
public interface RectangleRenderingInformation extends RenderingInformation {
	
	public void setPosition(Dimension2D position);
	
	public void setSize(Dimension2D size);
	
	public int gridXSize();
	
	public int gridYSize();
	
	public OPair<BigFraction> gridXPosition();
	
	public OPair<BigFraction> gridYPosition();
	
	public void setGridXPosition(OPair<BigFraction> gx);
	
	public void setGridYPosition(OPair<BigFraction> gy);
	
	public void setGridXSize(int x);
	
	public void setGridYSize(int y);
		
}

File: src/main/kotlin/org/kite9/diagram/model/position/RectangleRenderingInformation.kt
package org.kite9.diagram.model.position

import org.kite9.diagram.common.fraction.LongFraction
import org.kite9.diagram.common.objects.OPair

/**
 * Contains details of how to render a rectangle on screen, possibly containing some
 * text.
 *
 *
 * @author robmoffat
 */
interface RectangleRenderingInformation : RenderingInformation {

    fun gridXSize(): Int
    fun gridYSize(): Int
    fun gridXPosition(): OPair<LongFraction>?
    fun gridYPosition(): OPair<LongFraction>?
    fun setGridXPosition(gx: OPair<LongFraction>?)
    fun setGridYPosition(gy: OPair<LongFraction>?)
    fun setGridXSize(x: Int)
    fun setGridYSize(y: Int)
    var cellXPositions: DoubleArray?
    var cellYPositions: DoubleArray?
}